src/Database/WorkflowRepository.cs

```diff
@@ -0,0 +1,74 @@
+﻿// SPDX-FileCopyrightText: © 2021-2022 MONAI Consortium
+// SPDX-License-Identifier: Apache License 2.0
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Monai.Deploy.WorkflowManager.Database.Interfaces;
+using Monai.Deploy.WorkflowManager.Contracts.Models;
+using Monai.Deploy.WorkflowManager.Database.Options;
+using MongoDB.Driver;
+using System;
+
+namespace Monai.Deploy.WorkflowManager.Database
+{
+    public class WorkflowRepository : IWorkflowRepository
+    {
+        private readonly IMongoCollection<Workflow> _workflowCollection;
+
+        public WorkflowRepository(
+            IMongoClient client,
+            IOptions<WorkloadManagerDatabaseSettings> databaseSettings)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            var mongoDatabase = client.GetDatabase(databaseSettings.Value.DatabaseName);
+            _workflowCollection = mongoDatabase.GetCollection<Workflow>(databaseSettings.Value.WorkflowCollectionName);
+        }
+
+        public async Task<Workflow> GetByWorkflowIdAsync(string workflowId)
+        {
+            var workflow = await _workflowCollection
```

Comment:
may want to check for parameters here unless `workflowId` allows empty or null

Reply:
Added null checks